flink-table-store-core/src/main/java/org/apache/flink/table/store/file/operation/FileStoreExpireImpl.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.store.file.operation;
+
+import org.apache.flink.core.fs.Path;
+import org.apache.flink.table.data.binary.BinaryRowData;
+import org.apache.flink.table.store.file.Snapshot;
+import org.apache.flink.table.store.file.manifest.ManifestEntry;
+import org.apache.flink.table.store.file.manifest.ManifestFileMeta;
+import org.apache.flink.table.store.file.manifest.ManifestList;
+import org.apache.flink.table.store.file.mergetree.sst.SstPathFactory;
+import org.apache.flink.table.store.file.utils.FileStorePathFactory;
+import org.apache.flink.table.store.file.utils.FileUtils;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Default implementation of {@link FileStoreExpire}. It retains a certain number of latest
+ * snapshots.
+ */
+public class FileStoreExpireImpl implements FileStoreExpire {
+
+    private final int numRetained;
+    private final FileStorePathFactory pathFactory;
+    private final ManifestList manifestList;
+    private final FileStoreScan scan;
+
+    public FileStoreExpireImpl(
```

Comment:
I am thinking that we can provide three options:
- snapshot-expire.retention: retention time for snapshot changes
- snapshot-expire.num-retained
- snapshot-expire.check-interval

Reply:
`Expire` is an operation called by users (for example by connector sinks). `snapshot-expire.check-interval` should be checked by users, not by the operation itself.

As for retention time I'll implement this.

Reply:
Can you modify the caller too? `StoreGlobalCommitter`?